client/src/cbltest/api/syncgateway.py

```diff
@@ -1098,6 +1096,236 @@ async def get_document_revision_public(
+        local_path: str,
+        caddy_port: int = 20000,
+    ) -> None:
+        """
+        Downloads a file from the remote server via Caddy HTTP server
+
+        :param remote_filename: Name of the file on the remote server (e.g., 'sgcollectinfo-xxx-redacted.zip')
+        :param local_path: Local path where the file should be saved
+        :param caddy_port: Port where Caddy is serving files (default: 20000)
+        :raises FileNotFoundError: If the file doesn't exist
+        :raises Exception: For other HTTP errors
+        """
+        caddy_url = f"http://{self.hostname}:{caddy_port}/{remote_filename}"
+
+        with self._tracer.start_as_current_span(
+            "download_file_via_caddy",
+            attributes={
+                "cbl.remote.filename": remote_filename,
+                "cbl.local.path": local_path,
+                "cbl.caddy.url": caddy_url,
+            },
+        ):
+            _, content = await self._caddy_http_request(
+                caddy_url, f"Download {remote_filename}", timeout=120
+            )
+
+            # Ensure local directory exists and write file
+            local_file_path = Path(local_path)
+            local_file_path.parent.mkdir(parents=True, exist_ok=True)
+            local_file_path.write_bytes(content)
+
+            cbl_info(
+                f"Successfully downloaded {remote_filename} to {local_path} ({len(content)} bytes)"
+            )
+
+    async def list_files_via_caddy(
+        self,
+        caddy_port: int = 20000,
+        pattern: str | None = None,
+    ) -> list[str]:
+        """
+        Lists files available in the Caddy-served directory (requires 'browse' enabled in Caddyfile)
+
+        :param caddy_port: Port where Caddy is serving files (default: 20000)
+        :param pattern: Optional regex pattern to filter filenames (e.g., 'sgcollect_info.*redacted.zip')
+        :return: List of filenames available in the directory
+        :raises Exception: If directory browsing is not enabled or request fails
+        """
+        caddy_url = f"http://{self.hostname}:{caddy_port}/"
+
+        with self._tracer.start_as_current_span(
+            "list_files_via_caddy",
+            attributes={
+                "cbl.caddy.url": caddy_url,
+                "cbl.pattern": pattern or "all",
+            },
+        ):
+            try:
+                _, content = await self._caddy_http_request(
+                    caddy_url, "List directory", timeout=30
+                )
+            except FileNotFoundError:
+                raise Exception(
+                    "Directory browsing endpoint not found. "
+                    "Ensure Caddy is configured with 'file_server browse'"
+                )
+
+            html_content = content.decode("utf-8")
+
+            # Extract filenames from HTML anchor tags
+            # Caddy browse uses: <a href="./filename.ext">
+            href_pattern = re.compile(r'<a\s+href="\.?/([^"]+)"')
```

Comment:
Stop this, I sent you the docs for caddy to show how to get the results in json instead.  If you ever find yourself parsing HTML you are almost always doing something wrong.